quint/src/names/collector.ts

```diff
@@ -142,13 +144,14 @@ export class NameCollector implements IRVisitor {
       }
 
       // Update the definition to point to the expression being overriden
-      instanceTable.set(param.name, { ...constDef, id: ex.id })
+      constDef.id = ex.id
+      constDef.hidden = false
```

Comment:
So we don't have to update the `instanceTable` at all now? Or is that happening somewhere else now?

Reply:
I'm mutating the values themselves.
Before, we updated the whole value of the map (which was a single def). Now, the values are arrays, so it would be tricky to update the value without accidentally mutating the array. Therefore, I decided to `cloneDeep` the entire map, to be safe. And, since it is safe to mutate things, we can just mutate the definition directly.